pontoon/base/models/entity.py

```diff
@@ -866,22 +875,36 @@ def for_project_locale(
                 "id", flat=True
             )
 
-            # if not search_translations_only:
-            q_key = Q(key__icontains=search)  # if search_identifiers else Q()
-            entity_filters = (
-                Q(string__icontains=search) | Q(string_plural__icontains=search) | q_key
-                for search in search_list
-            )
+            # Search in source strings
+            if not search_translations_only:
+                # Search in string (context) identifiers
+
+                case_lookup = "contains" if search_match_case else "icontains"
```

Comment:
Can we simply use the `translation_case_lookup` here (and rename it to `case_lookup`)?

Reply:
Leaving in the `collate` and re-suing the same variable results in an error with `/get-entities/` for me :(

Reply:
Is the error still there if you add the 2nd parameter, i.e. `locale.db_collation`?

Reply:
OTOH, we have en-US hardcoded in a few places and collation is not needed for it, so let's keep it like this.

Reply:
![image](https://github.com/user-attachments/assets/ded593e7-dcfc-41c4-9ed7-1e9fbf9a613d)
![image](https://github.com/user-attachments/assets/06aeccd0-4c79-4e2f-ac50-c03fd07fdfc0)
<img width="409" alt="image" src="https://github.com/user-attachments/assets/d499a5bf-8515-4d8d-bb4c-b282eb7cb2c3">


It seems I no longer get any results when adding in the second argument

Reply:
(In this case, locale would need to be `Locale.objects.get(code="en-US")`).